doc/src/manual/noteworthy-differences.md

```diff
@@ -249,10 +252,12 @@ For users coming to Julia from R, these are some noteworthy differences:
   * The logical Julia program structure (Packages and Modules) is independent of the file structure (`include` for additional files), whereas the Python code structure is defined by directories (Packages) and files (Modules).
   * The ternary operator `x > 0 ? 1 : -1` in Julia corresponds to a conditional expression in Python `1 if x > 0 else -1`.
   * In Julia the `@` symbol refers to a macro, whereas in Python it refers to a decorator.
-  * Exception handling in Julia is done using `try` — `catch` — `finally`, instead of `try` — `except` — `finally`. In contrast to Python, it is not recommended to use exception handling as part of the normal workflow in Julia due to performance reasons.
+  * Exception handling in Julia is done using `try` — `catch` — `finally`, instead of `try` — `except` — `finally`. In contrast to Python, it is not recommended to use exception handling as part of the normal workflow in Julia (Julia's fast conditionals generally have much better performance than exception handling in either Julia or Python).
```

Comment:
This feels a bit unclear to me. What does fast conditionals have to do with exception handling? Is the reason one uses exceptions in Python because conditionals are slow?

Reply:
Yes, there are circumstances in Python where it's recommended to use exception handling for what we would consider ordinary execution: https://wiki.python.org/moin/PythonSpeed/PerformanceTips#Initializing_Dictionary_Elements

Reply:
Seems they recommend using the `get` method there.

Reply:
Yes, but the original advice that it's sometimes faster hasn't entirely gone away. The problem with how we phrased this initially is it leaves it ambiguous as to whether Julia's exception handling is slow or Python's ordinary execution is slow. I'm trying to clarify that it's the latter.

Reply:
Julia's exception handling is quite slow, no?

```python
In [1]: def f(x):
   ...:     try:
   ...:         y = 1/x
   ...:     except:
   ...:         y = 2.0
   ...:     return y
   ...:

In [2]: %timeit f(0)
235 ns ± 6.27 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

```jl
julia> function f(x)
           y = try
               div(1,x)
           catch
               2
           end
           return y
       end
f (generic function with 1 method)

julia> @btime f(0)
  15.963 μs (0 allocations: 0 bytes)
```